### PR TITLE
fix: Repair argument passing in both Dynamo paths

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Pass-through new TRT args in export
- Fix bug in `fx_ts_compat` where `explicit_batch_dimension` did not exist in input args
- Pass-through build failures arg in compile

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 1]
```diff
@@ -67,13 +71,17 @@ def compile(
...
+        explicit_batch_dimension: Whether to specify an explicit batch dimension to TRT
+        use_experimental_rt: Uses the next generation TRTModule which supports both Python and TorchScript based execution (including in C++).
+        max_aux_streams: max number of aux stream to use
+        version_compatible: enable version compatible feature
+        optimization_level: builder optimization level
     Returns:
         A torch.nn.Module lowered by TensorRT.
     """
-    if use_experimental_fx_rt and not explicit_batch_dimension:
+    if use_experimental_rt and not explicit_batch_dimension:
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:82]: Will this work since use_experimental_fx_rt was removed?

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 2]
```diff
@@ -122,7 +130,11 @@ def compile(
         save_timing_cache=save_timing_cache,
         cuda_graph_batch_size=cuda_graph_batch_size,
         is_aten=is_aten,
-        use_experimental_rt=use_experimental_fx_rt,
+        explicit_batch_dimension=explicit_batch_dimension,
+        use_experimental_rt=use_experimental_rt,
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:134]: similar thought here

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py (hunk 1 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:82]: `use_experimental_fx_rt ` was renamed to `use_experimental_rt` in the function header, for more consistent naming across the board (every other function uses `use_experimental_rt`).

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py]: I've since removed that change, since I noticed `torch_tensorrt.fx.compile(...)` still uses `use_experimental_fx_rt`, so it could be better to keep those the same

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 3]
```diff
@@ -67,7 +71,11 @@ def compile(
         timing_cache_prefix: Timing cache file name for timing cache used by fx2trt.
         save_timing_cache: Update timing cache with current timing cache data if set to True.
         cuda_graph_batch_size: Cuda graph batch size, default to be -1.
+        explicit_batch_dimension: Whether to specify an explicit batch dimension to TRT
         use_experimental_fx_rt: Uses the next generation TRTModule which supports both Python and TorchScript based execution (including in C++).
+        max_aux_streams: max number of aux stream to use
+        version_compatible: enable version compatible feature
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:77]: Can you let me know what version_compatible is and what it does?

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 4]
```diff
@@ -48,7 +48,11 @@ def compile(
     save_timing_cache=False,
     cuda_graph_batch_size=-1,
     is_aten=False,
+    explicit_batch_dimension=True,
```
[the remark this answers is not in the document]
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:51]: Do you think it should be removed from all of the FX code or only the Dynamo FX code?

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py (hunk 3 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:77]: This was introduced in #1935, but I am just ensuring users can call it from the frontend API. In the TRT documentation, [version_compatible](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/BuilderConfig.html?highlight=version_compatible) controls whether the engine plan file is forward-compatible.

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/fx_ts_compat/lower.py]: Removed all references to `explicit_batch_dimension` (deprecated to not use) and `explicit_precision` (has no effect), within the Dynamo code.

File: py/torch_tensorrt/dynamo/fx_ts_compat/lower.py [hunk 5]
```diff
@@ -198,7 +200,7 @@ def default_split_function(
     model: fx.GraphModule, inputs: Input, lower_setting: LowerSetting
 ) -> SplitResult:
     splitter_setting = TRTSplitterSetting()
-    splitter_setting.use_implicit_batch_dim = not lower_setting.explicit_batch_dimension
+    splitter_setting.use_implicit_batch_dim = False
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/fx_ts_compat/lower.py:203]: Since `explicit_batch_dimension` is no longer an allowed input argument due to deprecation, we set this to False.

**Review by peri044 — APPROVED** [review]
LGTM
